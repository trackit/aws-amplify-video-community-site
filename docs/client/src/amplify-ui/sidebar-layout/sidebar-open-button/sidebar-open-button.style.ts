import {css} from "emotion";
import {MQTablet} from "../../styles/media";

export const sidebarOpenButtonStyle = css`
  width: 2.75rem;
  margin-top: 1.875rem;
  padding: 0.75rem;
  position: absolute;

  border-top: 0.0625rem solid var(--border-color);
  border-right: 0.0625rem solid var(--border-color);
  border-bottom: 0.0625rem solid var(--border-color);
  border-radius: 0 0.3125rem 0.3125rem 0;
  box-shadow: rgba(0, 0, 0, 0.09) 0.08rem 0.15625rem 0 0.08rem;

  background-color: var(--bg-color);
  &:hover {
    background-color: var(--bg-color-hover);
    cursor: pointer;
  }

  display: none;
  ${MQTablet} {
    display: initial;
  }

  > a > img {
    width: 1.5rem;
    height: 1.5rem;
  }
`;
