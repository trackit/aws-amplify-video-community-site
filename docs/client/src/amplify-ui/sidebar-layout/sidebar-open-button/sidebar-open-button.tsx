import {Component, Prop, h, Host, State} from "@stencil/core";
import {sidebarLayoutContext} from "../sidebar-layout.context";
import {ToggleInView} from "../sidebar-layout.types";
import {sidebarOpenButtonStyle} from "./sidebar-open-button.style";
import {DESKTOP_OPEN} from "../../../constants/img";

@Component({tag: "amplify-sidebar-open-button"})
export class AmplifySidebarOpenButton {
  /*** toggles the state provided by `sidebar-layout` */
  @Prop() readonly toggleInView?: ToggleInView;
  /*** whether or not the sidebar is in view, provided by `sidebar-layout` */
  @Prop() readonly inView?: boolean;
  /*** whether or not the button is being hovered */
  @State() isHovered = false;

  onClick = (e: Event) => {
    if (this.inView) {
      e.stopPropagation();
    } else {
      // if we don't do this, the open icon is dark when the user closes the menu even if they are not hovering over it
      this.isHovered = false;
    }
    if (this.toggleInView) {
      this.toggleInView();
    }
  };

  hover = () => {
    this.isHovered = true;
  };

  unhover = () => {
    this.isHovered = false;
  };

  render() {
    const imgLink = this.isHovered
      ? DESKTOP_OPEN.darkSrc
      : DESKTOP_OPEN.lightSrc;
    return (
      !this.inView && (
        <Host>
          <button
            class={sidebarOpenButtonStyle}
            onClick={this.onClick}
            onMouseEnter={this.hover}
            onMouseLeave={this.unhover}
          >
            <img alt={DESKTOP_OPEN.alt} src={imgLink} />
          </button>
        </Host>
      )
    );
  }
}

sidebarLayoutContext.injectProps(AmplifySidebarOpenButton, [
  "inView",
  "toggleInView",
]);
