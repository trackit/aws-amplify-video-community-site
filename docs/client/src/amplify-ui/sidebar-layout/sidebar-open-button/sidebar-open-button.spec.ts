import {AmplifySidebarOpenButton} from "./sidebar-open-button";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-sidebar-layout-toggle", () => {
  describe("Component logic", () => {
    let amplifySidebarOpenButton: AmplifySidebarOpenButton;
    beforeEach(
      () => (amplifySidebarOpenButton = new AmplifySidebarOpenButton()),
    );

    it("should init `toggleInView` as `undefined`", () =>
      expect(amplifySidebarOpenButton.toggleInView).toBeUndefined());

    it("should init `inView` as `undefined`", () =>
      expect(amplifySidebarOpenButton.inView).toBeUndefined());

    it("should init `isHovered` as `false`", () =>
      expect(amplifySidebarOpenButton.isHovered).toBe(false));
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifySidebarOpenButton],
            html: `<amplify-sidebar-open-button />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
