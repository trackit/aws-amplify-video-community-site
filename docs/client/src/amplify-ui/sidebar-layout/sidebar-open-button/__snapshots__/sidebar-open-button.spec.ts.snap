// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sidebar-layout-toggle Render logic should render 1`] = `
<amplify-sidebar-open-button>
  <button class="css-1crc5v5">
    <img alt="Open menu" src="/assets/burger-light.svg">
  </button>
</amplify-sidebar-open-button>
`;
